test/integration/controller/core/clusterqueue_controller_test.go

```diff
@@ -708,28 +709,44 @@ var _ = ginkgo.Describe("ClusterQueue controller", ginkgo.Ordered, ginkgo.Contin
 
 	ginkgo.When("Deleting clusterQueues", func() {
 		var (
-			cq *kueue.ClusterQueue
-			lq *kueue.LocalQueue
+			cq    *kueue.ClusterQueue
+			lq    *kueue.LocalQueue
+			check *kueue.AdmissionCheck
 		)
 
 		ginkgo.BeforeEach(func() {
-			cq = testing.MakeClusterQueue("foo-cq").Obj()
+			check = testing.MakeAdmissionCheck("check").ControllerName("check-controller").Obj()
+			gomega.Expect(k8sClient.Create(ctx, check)).To(gomega.Succeed())
+
+			cq = testing.MakeClusterQueue("foo-cq").AdmissionChecks(check.Name).Obj()
 			lq = testing.MakeLocalQueue("queue", ns.Name).ClusterQueue(cq.Name).Obj()
 			gomega.Expect(k8sClient.Create(ctx, lq)).To(gomega.Succeed())
 			gomega.Expect(k8sClient.Create(ctx, cq)).To(gomega.Succeed())
 		})
 
+		ginkgo.AfterEach(func() {
+			util.ExpectAdmissionCheckToBeDeleted(ctx, k8sClient, check, true)
+		})
+
 		ginkgo.It("Should delete clusterQueues successfully when no admitted workloads are running", func() {
 			util.ExpectClusterQueueToBeDeleted(ctx, k8sClient, cq, true)
 		})
 
 		ginkgo.It("Should be stuck in termination until admitted workloads finished running", func() {
+			util.SetAdmissionCheckActive(ctx, k8sClient, check, metav1.ConditionTrue)
 			util.ExpectClusterQueueStatusMetric(cq, metrics.CQStatusActive)
 
 			ginkgo.By("Admit workload")
 			wl := testing.MakeWorkload("workload", ns.Name).Queue(lq.Name).Obj()
 			gomega.Expect(k8sClient.Create(ctx, wl)).To(gomega.Succeed())
 			gomega.Expect(util.SetQuotaReservation(ctx, k8sClient, wl, testing.MakeAdmission(cq.Name).Obj())).To(gomega.Succeed())
+			util.SetWorkloadsAdmissionCkeck(ctx, k8sClient, wl, check.Name, kueue.CheckStateReady, true)
+			gomega.Eventually(func(g gomega.Gomega) {
```

Comment:
We didn't get the method for WaitForWorkloadToBeAdmitted back? If not, we can leave it for a follow up.

Reply:
it's in #1241